### PR TITLE
.NET CLI support

### PR DESCRIPTION
**Pull request by liron-shalom**

### CODE REVIEW

File: artifactory/utils/dotnet/toolchaincmd.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+func (toolchainType ToolchainType) GetTypeFlagPrefix() string {
+	return CmdFlagPrefixes[toolchainType]
+}
+
+var AddSourceArgs = [][]string{
+	{"sources", "add"},
+	{"nuget", "add", "source"},
+}
```
Comment (eyalbe4) [inline artifactory/utils/dotnet/toolchaincmd.go:38]: We are maintaining here several arrays, with two elements, one of each toolchain. All of these arrays are loosely bound to the ToolchainType consts. If the elements order in any of these arrays is flipped, a bug will be created.
I suggest taking a more object oriented approach, and use a Toolchain struct, whch will include a field for each of the tool chain characteristics. We'll need to two Toolchain structs - one for Nuget and one for DotnetCore.

File: artifactory/utils/dotnet/toolchaincmd.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	addSourceCmd := Cmd{toolchain: cmdType, execPath: execPath, Command: cmdType.GetAddSourceArgs()}
+	switch cmdType {
+	case Nuget:
+		addSourceCmd.CommandFlags = append(addSourceCmd.CommandFlags, "-source", sourceUrl)
+	case Dotnet:
+		addSourceCmd.Command = append(addSourceCmd.Command, sourceUrl)
+		// Dotnet cli does not support password encryption on non-Windows os, so we will write the raw password.
```
Comment (eyalbe4) [inline artifactory/utils/dotnet/toolchaincmd.go:63]: ...non-Windows os... --> ...non-Windows OS...

File: docs/artifactory/dotnet/help.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package dotnet
+
+const Description = "Run .NET Core CLI"
+
+var Usage = []string{`jfrog rt dotnet <dotnet command> [command options]`}
```
Comment (eyalbe4) [inline docs/artifactory/dotnet/help.go:5]: dotnet command --> dotnet sub-command

Let's also do the same change in the pip command help:
pip command --> pip sub-command

**Comment by liron-shalom** [comment]
@eyalbe4,
I made all the requested structural changes except the last one, since removing the dotnetcommand package (former commandargs package) cause an import cycle.

File: artifactory/commands/utils/configfile.go [hunk 4]
```diff
@@ -89,6 +89,8 @@ func CreateBuildConfig(c *cli.Context, confType utils.ProjectType) (err error) {
 			err = configFile.configPip()
 		case utils.Npm:
 			err = configFile.configNpm()
+		case utils.Dotnet:
+			fallthrough
```
[the remark this answers is not in the document]
Comment (liron-shalom) [reply artifactory/commands/utils/configfile.go:93]: a [_fallthrough_ statement ](https://github.com/golang/go/wiki/Switch#fall-through)"transfers control to the next case", so only the next case will be executed in this case.
Do you still think it is better to call configFile.configDotnet() separatly on each case?

File: artifactory/utils/dotnet/commandargs/commandargs.go [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+	dca.args = args
+	return dca
+}
+
+func (dca *DotnetCommandArgs) RtDetails() (*config.ArtifactoryDetails, error) {
+	return dca.rtDetails, nil
+}
+
+func (dca *DotnetCommandArgs) CommandName() string {
+	return "rt_" + dca.toolchainType.String()
```
[the remark this answers is not in the document]
Comment (liron-shalom) [reply artifactory/utils/dotnet/commandargs/commandargs.go]: The values in `CmdTypes` are lower case, so the command name is lower case as well.

File: artifactory/commands/utils/configfile.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (eyalbe4) [reply artifactory/commands/utils/configfile.go:93]: In that case, your code is good.

File: artifactory/utils/projectconfig.go [hunk 6]
```diff
@@ -133,3 +135,25 @@ func (repo *RepositoryConfig) SetRtDetails(rtDetails *config.ArtifactoryDetails)
...
+	// Get configuration file path.
+	confFilePath, exists, err := GetProjectConfFilePath(projectType)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errorutils.CheckError(fmt.Errorf(projectType.String() + " Project configuration does not exist."))
+	}
+	return ReadResolutionOnlyConfiguration(confFilePath)
+
```
Comment (eyalbe4) [inline artifactory/utils/projectconfig.go:149]: Redundan new-line here.

File: artifactory/cli.go [hunk 7]
```diff
@@ -2199,7 +2257,52 @@ func nugetDepsTreeCmd(c *cli.Context) error {
...
+	filteredDotnetArgs, buildConfiguration, err := utils.ExtractBuildDetailsFromArgs(args)
+	if err != nil {
+		return err
+	}
+
+	// Run command.
+	dotnetCmd := dotnet.NewDotnetCoreCliCommand()
+	dotnetCmd.SetRtDetails(rtDetails).SetRepoName(dotnetConfig.TargetRepo()).SetBuildConfiguration(buildConfiguration).SetArgs(filteredDotnetArgs[0])
+	if len(filteredDotnetArgs) > 1 {
+		dotnetCmd.SetFlags(strings.Join(filteredDotnetArgs[1:], " "))
```
Comment (eyalbe4) [inline artifactory/cli.go:2303]: Let's add a comment describing this.

**Pull request merged**